mlflow/models/utils.py

```diff
@@ -84,6 +89,16 @@ def __init__(self, input_example: ModelInputExample):
         def _is_scalar(x):
             return np.isscalar(x) or x is None
 
+        def _contains_params(x):
+            return isinstance(x, dict) and EXAMPLE_PARAMS_KEY in x
```

Comment:
Instead of `dict`, can we use tuple like this?

```python
log_model(..., input_example=(example, params)
```

Reply:
With dict, it's a pain to split it into example and parasm. With tuple, we can just unpack. No need to add a special key for params.

Reply:
Yea that's also one option. Then users won't be able to pass tuple as input in the future if we want to support it.

Reply:
> Then users won't be able to pass tuple as input in the future if we want to support it.

This sounds fine to me. They can still use `list`.